src/utils/all_reduce.hpp

```diff
@@ -0,0 +1,106 @@
+//========================================================================================
+// (C) (or copyright) 2021. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+#ifndef UTILS_ALL_REDUCE_HPP_
+#define UTILS_ALL_REDUCE_HPP_
+
+#include <vector>
+
+#include <config.hpp>
+#include <parthenon_mpi.hpp>
+
+namespace parthenon {
+
+enum class Op { MAX, MIN, SUM, PROD, LAND, BAND, LOR, BOR, LXOR, BXOR, MAXLOC, MINLOC };
+#ifndef MPI_PARALLEL
+#define MPI_
+#endif
```

Comment:
This is both clever and evil.

Reply:
Mwahahahaha.  Haven't even tested that it works, to be honest.

Reply:
Wow. Make sure it works on a few different compilers when you do check

Reply:
I updated how I dealt with this.  It's less insane now, and works for serial builds..